app.js

```diff
@@ -0,0 +1,321 @@
+    
+    const elements = {
+        loader: document.getElementById("loader"),
+        coinsContainer: document.getElementById("coins-container"),
+        reportsContainer: document.getElementById("reports-container"),
+        aboutContainer: document.getElementById("about-container"),
+        chartContainer: document.getElementById("chartContainer"),
+        searchInput: document.getElementById("search-input"),
+        searchForm: document.getElementById("search-form"),
+        modalElement: document.getElementById('replaceCoinModal'),
+        modalBody: document.getElementById('selected-coins-list'),
+    };
+   
+    const replaceCoinModal = new bootstrap.Modal(elements.modalElement);
+
+    
+
+    
+    async function fetchInitialCoins() {
+        showLoader(true);
+        try {
+            const response = await fetch(API_URLS.allCoins);
+            if (!response.ok) throw new Error("Network response was not ok");
+            const coins = await response.json();
+            state.allCoins = coins;
+            displayCoins(coins); 
+        } catch (error) {
+            elements.coinsContainer.innerHTML = `<div class="alert alert-danger">שגיאה בטעינת המידע.</div>`;
+            console.error("Failed to fetch coins:", error);
+        } finally {
+            showLoader(false);
+        }
+    }
+
+    
+    function displayCoins(coinsToDisplay) {
+        
+        if (coinsToDisplay.length === 0) {
+            elements.coinsContainer.innerHTML = `<div class="alert alert-warning">לא נמצאו מטבעות.</div>`;
+            return;
+        }
+
+       
+        const allCardsHTML = coinsToDisplay.map(coin => {
+            
+            const isChecked = state.selectedForReport.includes(coin.symbol.toUpperCase());
+            return `
+                <div class="col-sm-6 col-md-4 col-lg-3 mb-4">
+                    <div class="card coin-card">
+                        <div class="card-body">
+                             <div class="form-check form-switch float-end">
+                                <input class="form-check-input report-toggle" type="checkbox" 
+                                       data-coin-symbol="${coin.symbol.toUpperCase()}" ${isChecked ? 'checked' : ''}>
+                            </div>
+                            <h5 class="card-title">${coin.symbol.toUpperCase()}</h5>
+                            <p class="card-text">${coin.name}</p>
+                            <button class="btn btn-primary more-info-btn" data-coin-id="${coin.id}">More Info</button>
+                            <div class="collapse" id="collapse-${coin.id}">
+                               <div class="more-info-area mt-3">
+                                   </div>
+                            </div>
+                        </div>
+                    </div>
+                </div>`;
+        }).join('');
+        elements.coinsContainer.innerHTML = allCardsHTML;
+    }
+
+   
+    async function handleMoreInfoClick(coinId) {
```

Comment:
it should have hovered... 

<img width="1723" height="564" alt="Screenshot 2025-09-10 at 20 51 49" src="https://github.com/user-attachments/assets/5e86dff3-fbb1-43c2-89aa-d4a2f2f837be" />